### PR TITLE
Fixed to compile errors

### PR DESCRIPTION
**Pull request by beatrizleon**
This package was giving errors because of the order of an include, which was solved. Also improved the README file.

### CODE REVIEW

File: urdf_to_simox_xml/README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
-How to run the URDF to Simox XML converter.
+It needs to have meshlab installed: 
+```
+sudo apt-get install meshlab
```
Comment (ugocupcic) [inline urdf_to_simox_xml/README.md]: add meshlab to the rosdep dependencies in package.xml instead?

File: urdf_to_simox_xml/README.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+```
 
+How to run the URDF to Simox XML converter.
+```
 $ cd ~/catkin_ws
```
Comment (ugocupcic) [inline urdf_to_simox_xml/README.md]: remove `$` at the beginning of the lines for easy copy paste

File: urdf_to_simox_xml/README.md [hunk 3]
```diff
@@ -9,21 +14,23 @@ $ catkin_make
...
-
+```
 ---------
+The output_dir should point to the path where the *model* folder with the .wrl files are located. For example, to generate the model of the shadow hand:
+rosrun urdf_to_simox_xml urdf_to_simox_xml --output_dir simox_ros/sr_grasp_description/simox
 
 Use RobotViewer to verify the output (xml files such as shadowhand.xml and dms.xml):
-
+```
 $ RobotViewer
```
Comment (ugocupcic) [inline urdf_to_simox_xml/README.md]: same here

File: urdf_to_simox_xml/README.md [hunk 4]
```diff
@@ -9,21 +14,23 @@ $ catkin_make
...
-
+```
 Click on the "Select Robot File" button on the panel, and select the xml file.
 
 Note that RobotViewer is part of http://simox.sourceforge.net. If you cannot find it, make sure that you have installed SoQt library before installing Simox.
 
 To grasp an object with the hand (e.g., the DMS hand), run GraspPlanner (also part of Simox).
-
+```
 $ GraspPlanner --robot dms.xml --endeffector "DMS Hand" --preshape "Grasp Preshape"
```
Comment (ugocupcic) [inline urdf_to_simox_xml/README.md]: same here

File: urdf_to_simox_xml/src/urdf_to_simox_xml.cpp [hunk 5]
```diff
@@ -23,7 +23,6 @@
 #include <string>
 #include <sstream>
 #include <vector>
-#include <boost/foreach.hpp>
```
Comment (ugocupcic) [inline urdf_to_simox_xml/src/urdf_to_simox_xml.cpp]: why did you move that to the header? It's better to keep the dependencies in the header to a minimum.

File: urdf_to_simox_xml/README.md (hunk 1 above)
Comment (beatrizleon) [inline urdf_to_simox_xml/README.md]: Ok

File: urdf_to_simox_xml/src/urdf_to_simox_xml.cpp (hunk 5 above)
Comment (beatrizleon) [inline urdf_to_simox_xml/src/urdf_to_simox_xml.cpp]: I was getting the following errors:
[ 95%] Building CXX object shadow_robot_ethercat/sr_robot_lib/CMakeFiles/sr_hand_lib.dir/src/motor_updater.cpp.o
In file included from /home/beatriz/workspace/shadow/src/simox_ros/urdf_to_simox_xml/src/urdf_to_simox_xml.cpp:26:0:
/usr/include/boost/foreach.hpp:169:42: error: ‘boost::Q_FOREACH::tag’ has not been declared
 boost_foreach_is_lightweight_proxy(T *&, BOOST_FOREACH_TAG_DEFAULT) { return 0; }
                                          ^
/usr/include/boost/foreach.hpp:173:72: error: ‘boost::Q_FOREACH::tag’ has not been declared
 boost_foreach_is_lightweight_proxy(std::pair<T, T> *&, boost::foreach::tag) { return 0; }
                                                                        ^
/usr/include/boost/foreach.hpp:177:81: error: ‘boost::Q_FOREACH::tag’ has not been declared
 boost_foreach_is_lightweight_proxy(boost::iterator_range<T> *&, boost::foreach::tag) { return 0; }
                                                                                 ^
/usr/include/boost/foreach.hpp:181:76: error: ‘boost::Q_FOREACH::tag’ has not been declared
 boost_foreach_is_lightweight_proxy(boost::sub_range<T> *&, boost::foreach::tag) { return 0; }
                                                                            ^
/usr/include/boost/foreach.hpp:185:59: error: ‘boost::Q_FOREACH::tag’ has not been declared
 boost_foreach_is_lightweight_proxy(T **&, boost::foreach::tag) { return 0; }
                                                           ^
/usr/include/boost/foreach.hpp:194:36: error: ‘boost::Q_FOREACH::tag’ has not been declared
 boost_foreach_is_noncopyable(T *&, BOOST_FOREACH_TAG_DEFAULT) { return 0; }

I found that the boost libraries should be included before the Qt, which were included in the header. Moving it there before solved the issue. How do you suggest me to do it instead?

Comment (ugocupcic) [inline urdf_to_simox_xml/src/urdf_to_simox_xml.cpp]: Ah! OK no worries in that case.
